File: Leetcode 3701-3800/3747/3747.java
// Leetcode 3747: Count Distinct Integers After Removing Zeros
// https://leetcode.com/problems/count-distinct-integers-after-removing-zeros/
// Solved on 21st of November, 2025
class Solution {
    /**
     * Counts the number of distinct integers after removing zeros.
     * @param n The input long integer.
     * @return The count of distinct integers.
     */
    public long countDistinct(long n) {
        String numString = Long.toString(n);
        int length = numString.length();
        long count = 0;
        long power = 1;

        for (int i = 1; i < length; i++) {
            power *= 9;
            count += power;
        }

        for (int i = 0; i < length; i++) {
            int digit = numString.charAt(i) - '0';
            if (digit > 1) {
                count += (digit - 1) * power;
            }
            if (digit == 0) {
                return count;
            }
            power /= 9;
        }

        return count + 1;
    }
}

File: Leetcode 3701-3800/3747/3747.py
# Leetcode 3747: Count Distinct Integers After Removing Zeros
# https://leetcode.com/problems/count-distinct-integers-after-removing-zeros/
# Solved on 21st of November, 2025
class Solution:
    def countDistinct(self, n: int) -> int:
        """
        Counts the number of distinct integers that can be formed by removing leading zeros from numbers
        less than or equal to n.
        :param n: The upper bound integer.
        :return: The total count of distinct integers.
        """
        numString = str(n)
        numLength = len(numString)
        totalCount = 0

        currentPower = 9
        for i in range(numLength - 1):
            totalCount += currentPower
            currentPower *= 9

        for i in range(numLength):
            digit = int(numString[i])
            remainingDigits = numLength - 1 - i

            if digit == 0:
                return totalCount

            totalCount += (digit - 1) * (9 ** remainingDigits)

        return totalCount + 1